### PR TITLE
Change idiom from e.g. `shuffle` to `toShuffled` and `shuffleInPlace` to `shuffle`

### PR DESCRIPTION
**Pull request by chenglou**
And do the same for `fill`

See https://github.com/rescript-association/rescript-core/pull/139#issuecomment-1557231008 and https://github.com/rescript-association/rescript-core/pull/139#issuecomment-1558958913

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
+
+- `Array` mutable & immutable helper name changed to conform to JS' upcoming APIs [such as `toSorted`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted)
+  - `sort` -> `toSorted`, `sortInplace` -> `sort`
+  - `reverse` -> `toReversed`, `reverseInplace` -> `reverse`
+  - `splice` -> `toSpliced`, `spliceInPlace` -> `splice`
+  - `shuffle` -> `toShuffled`, `shuffleInPlace` -> `shuffle`
+  - `fillAllInPlace` -> `fillAll`, `fillInPlaceToEnd` -> `fillToEnd`, `fillInPlace` -> `fill`
+  - added `with`
+- Same for `TypedArray`:
+  - `sort` -> `toSorted`, `sortInplace` -> `sort`
```
Comment (cknitt) [inline CHANGELOG.md:17]:
```suggestion
  - `sort` -> `toSorted`, `sortInPlace` -> `sort`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+  - `reverse` -> `toReversed`, `reverseInplace` -> `reverse`
```
Comment (cknitt) [inline CHANGELOG.md:18]:
```suggestion
  - `reverse` -> `toReversed`, `reverseInPlace` -> `reverse`
```

File: README.md [hunk 3]
```diff
@@ -111,7 +111,7 @@ This standard library is based on `rescript-js`, but with the tweaks and modific
 - `setUnsafe` added (copied from `Belt`).
 - `sort`, `toSorted`, `reverse`, `toReversed`, `splice`, `toSpliced` are the same as their JS counterpart (mutable and immutable, respectively).
 - `keepMap` is added from `Belt`, but **renamed to `filterMap`**. Rationale: `filterMap` is closer to the JS convention of naming. It's also available in other languages like Rust. `keep` et al can confuse beginners, who're bound to be looking for `filter` style names since that's what JS has.
-- `shuffle` and `shuffleInPlace` are added (copied from `Belt`).
+- `shuffle` and `toShuffled` are added (copied from `Belt`'s `shuffleInplace` and `shuffle`).
```
Comment (cknitt) [inline README.md:114]:
```suggestion
- `shuffle` and `toShuffled` are added (copied from `Belt`'s `shuffleInPlace` and `shuffle`).
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,28 @@
 
 ## main
```
Comment (zth) [inline CHANGELOG.md:3]: Can remove this line as well if we're adding `Next version`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,28 @@
...
+  - `reverse` -> `toReversed`, `reverseInPlace` -> `reverse`
+  - `splice` -> `toSpliced`, `spliceInPlace` -> `splice`
+  - `shuffle` -> `toShuffled`, `shuffleInPlace` -> `shuffle`
+  - `fillAllInPlace` -> `fillAll`, `fillInPlaceToEnd` -> `fillToEnd`, `fillInPlace` -> `fill`
+  - added `with`
+- Same for `TypedArray`:
+  - `sort` -> `toSorted`, `sortInPlace` -> `sort`
+  - `reverse` -> `toReversed`, `reverseInPlace` -> `reverse`
+  - `fillAllInPlace` -> `fillAll`, `fillInPlaceToEnd` -> `fillToEnd`, `fillInPlace` -> `fill`
+- And `List:
```
Comment (cknitt) [inline CHANGELOG.md:20]:
```suggestion
- And `List`:
```

**Comment by chenglou** [comment]
Sorry for the typos
cc @glennsl

**Comment by chenglou** [comment]
> could we add a more elaborate note on the naming conventions

Feels like it's covered by the first point? Seeing it, it'd be normal to expect `reverse` and `toReversed` to be called exactly that (which semi-validates the changes here)

File: src/Core__Array.res [hunk 6]
```diff
@@ -7,18 +7,18 @@ external setUnsafe: (array<'a>, int, 'a) => unit = "%array_unsafe_set"
...
 external fromArrayLikeWithMap: (Js.Array2.array_like<'a>, 'a => 'b) => array<'b> = "Array.from"
 
-@send external fillAllInPlace: (array<'a>, 'a) => unit = "fill"
+@send external fillAll: (array<'a>, 'a) => unit = "fill"
 
-@send external fillInPlaceToEnd: (array<'a>, 'a, ~start: int) => unit = "fill"
+@send external fillToEnd: (array<'a>, 'a, ~start: int) => unit = "fill"
 
-@send external fillInPlace: (array<'a>, 'a, ~start: int, ~end: int) => unit = "fill"
+@send external fill: (array<'a>, 'a, ~start: int, ~end: int) => unit = "fill"
```
Comment (chenglou) [inline src/Core__Array.res:14]: Once the uncurry matures, these will all be subsumed into `fill`, without trailing `()`. The our one-off `toEnd` suffixes can all be (gently) deprecated too.

**Review by glennsl — APPROVED** [review]

**Pull request merged**